packages/nx-go/src/migrations/update-3.1.0/add-tidy-target-to-libraries-and-applications.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  formatFiles,
+  getProjects,
+  type Tree,
+  updateProjectConfiguration,
+} from '@nx/devkit';
+
+/**
+ * Update executor options to ensure a smooth transition to v3.
+ *
+ * @param tree the project tree
+ */
+export default async function update(tree: Tree) {
+  const projects = getProjects(tree);
+
+  for (const [projectName, projectConfig] of projects) {
+    let shouldUpdate = false;
+
+    if (!projectConfig.targets) continue;
+
+    if (
+      !projectConfig.targets.tidy &&
+      projectConfig.targets.tidy?.executor !== '@nx-go/nx-go:tidy'
```

Comment:
I have a question about this migration: does it manage non-Go projects in an Nx workspace? For example if I have back and front apps, I do not want this migration to add a tidy executor for the front app.

Shouldn't we also test if there we are in a Go workspace?

Reply:
Good catch I totally missed that. Will add it